fitting_source/1D_fitting/plot_6_139La_Ext_Czjzek.py

```diff
@@ -0,0 +1,276 @@
+from mrsimulator import signal_processor as sp
+
+import numpy as np
+import csdmpy as cp
+import lmfit
+from lmfit import Minimizer
+import matplotlib.pyplot as plt
+
+
+# sphinx_gallery_thumbnail_number = 3
+
+# %%
+# Import the dataset
+# ------------------
+host = "http://ssnmr.org/sites/default/files/"
+filename = "mrsimulator/La bc - LaY90 cpmg.csdf"
+experiment = cp.load(host + filename).real
+experiment.x[0].to("ppm", "nmr_frequency_ratio")
+experiment /= experiment.max()
+
+plt.figure(figsize=(6, 4))
+ax = plt.subplot(projection="csdm")
+ax.plot(experiment, "k", alpha=0.5)
+plt.grid()
+plt.tight_layout()
+plt.show()
+
+# %%
+# Calculate Noise Standard Deviation
+# ----------------------------------
+loc = np.where(experiment.dimensions[0].coordinates < -5000e-6)
+sigma = experiment[loc].std()
+print(sigma)
+
+# %%
+# Setup Method and Processor
+# --------------------------
+#
+# Next setup a method which simulates the experiment, again using the
+# :py:meth:`~mrsimulator.utils.get_spectral_dimensions` function to get the spectral
+# dimension parameters from the experimental dataset.
+# We also create a signal processor object which will scale the simulated spectrum.
+spec_dims = get_spectral_dimensions(experiment)
+method = BlochDecayCTSpectrum(
+    channels=["139La"], magnetic_flux_density=17.6, spectral_dimensions=spec_dims
+)
+
+processor = sp.SignalProcessor(operations=[sp.Scale(factor=1000)])
+
+
+# %%
+# Create a Lineshape Kernel
+# -------------------------
+#
+# An approach for fitting model values to the experimental dataset could be computing
+# the spectrum from a list of spin system objects with tensor parameters pulled from
+# an Extended Czjzek model during each round of fitting, but this would be inefficient!
+# The grid of :math:`\text{Cq}`
+# and :math:`\eta_q` points will not change during the fit; we are interested in the
+# probability distribution across these points.
+#
+# We therefore define a function which pre-computes a lineshape kernel given a range
+# of :math:`\text{Cq}` and :math:`\eta_q` values, an NMR method, and an isotope which,
+# for this example, is ``"139La"``.
+def make_kernel(cq_range, eta_range, method, isotope):
+    """Pre-computes the kernel to use when fitting the experimental spectrum
+
+    Arguments:
+        (np.array) cq_range: Numpy array of cq points
+        (np.array) eta_range: Numpy array of eta points
+        (Method) method: mrsimulator Method object used to simulate the spectra
+        (str) isotope: Isotope of the site
+
+    Returns:
+        Lineshape kernel as a numpy array
+    """
+    Cq, eta = np.meshgrid(cq_range, eta_range, indexing="xy")
+
+    spin_systems = [
+        SpinSystem(sites=[Site(isotope=isotope, quadrupolar=dict(Cq=cq * 1e6, eta=e))])
+        for cq, e in zip(Cq.flatten(), eta.flatten())
+    ]
+    sim = Simulator(spin_systems=spin_systems, methods=[method])
+    sim.config.decompose_spectrum = "spin_system"
+    sim.run(pack_as_csdm=False)
```

Comment:
Add a line that `pack_as_csdm` returns a numpy array instead of the default csdm object.